gradebookng/impl/src/main/java/org/sakaiproject/grading/impl/GradingServiceImpl.java

```diff
@@ -2548,14 +2548,28 @@ public Double convertStringToDouble(final String doubleAsString) {
      * shared GradebookAssignment object.
      *
      * @param gradebookUid
+     * @param siteId
      * @param categoryName
      * @return
      */
     private List<Assignment> getAssignmentsWithCategory(String gradebookUid, String siteId, String categoryName) {
```

Comment:
change this to take a category id instead of name. 

then the db query is even faster and is not doing a non-indexed search against `name`

Reply:
Changed. Thanks!